File: phoneBooks/src/phoneBooks/Telephone.java
package phoneBooks;

import java.util.ArrayList;
import java.util.List;

public class Telephone {
    private List<String> telephones = new ArrayList<>();

    public Telephone(String t) {
        telephones.add(t);
    }

    public Telephone() {}

    public List<String> getTelephones() {
        return telephones;
    }

    public boolean addTelephone(String t) {
        if (isTelephone(t)) {
            return false;
        }
        telephones.add(t);
        return true;
    }

    public boolean isTelephone(String t) {
        return telephones.contains(t);
    }

    public int phoneCount() {
        return telephones.size();
    }

    public boolean removePhone(String t) {
        if (telephones.contains(t)) {
            telephones.remove(t);
            return true;
        }
        return false;
    }

    @Override
    public boolean equals(Object o) {
        Telephone other = (Telephone) o;
        for (String s : other.getTelephones()) {
            if (!isTelephone(s)) {
                return false;
            }
        }
        return true;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (String s : telephones) {
            sb.append(String.format("%30s",s)).append("\n");
        }
        return sb.toString();
    }

    public String phonesToFile() {
        StringBuilder sb = new StringBuilder();
        for (String s : telephones) {
            sb.append(s).append(";");
        }
        sb.deleteCharAt(sb.length() - 1);
        return sb.toString();
    }
}

File: phoneBook/phoneBook/Telephone.cs
using phoneBook.bookIOFile;

namespace phoneBook.phoneBook
{
    public class Telephone
    {
        public List<string> Telephones
        {
            get => telephones;
            set { }
        }
        private List<string> telephones = new List<string>();

        public Telephone() { }

        public Telephone(string t) => telephones.Add(t);

        public Telephone(List<string> list) => telephones.AddRange(list);

        public bool AddTelephone(string t)
        {
            if (telephones.Contains(t))
            {
                List<string> strList = new List<string>();
                strList.Add($"тел. {t} уже есть!");
                ErrorConsole.Message(strList);
                return false;
            }
            else
            {
                telephones.Add(t);
                return true;
            }
        }

        public bool RemovePhone(string t)
        {
            if (telephones.Contains(t))
            {
                telephones.Remove(t);
                return true;
            }
            return false;
        }


        public bool IsPhone(string t)
        {
            return telephones.Contains(t);
        }

        public int Count => telephones.Count;

        public string PhonesToFile()
        {
            string str = string.Empty;
            foreach (string s in telephones)
            {
                str += s + ";";
            }
            return str.Remove(str.Length - 1);
        }

        public override bool Equals(object? obj)
        {
            if (obj == null || !this.GetType().Equals(obj.GetType()))
            {
                return false;
            }
            Telephone t = (Telephone)obj;
            foreach (string s in t.telephones)
            {
                if (!this.telephones.Contains(s))
                {
                    return false;
                }
            }
            return true;
        }

        public override int GetHashCode()
        {
            int result = 0;
            foreach (string s in telephones)
            {
                result += s.GetHashCode();
            }
            return result;
        }

        public override string ToString()
        {
            string str = string.Empty;
            foreach (string s in telephones)
            {
                str += String.Format("тел. {0}\n", s);
            }
            return str;
        }
    }
}